src/Npgsql/Util/Statics.cs

```diff
@@ -10,16 +10,19 @@ namespace Npgsql.Util;
 static class Statics
 {
 #if DEBUG
+    internal static bool Diagnostics;
     internal static bool LegacyTimestampBehavior;
     internal static bool DisableDateTimeInfinityConversions;
 #else
+    internal static readonly bool EnableDiagnostics;
     internal static readonly bool LegacyTimestampBehavior;
     internal static readonly bool DisableDateTimeInfinityConversions;
 #endif
 
     static Statics()
     {
-        LegacyTimestampBehavior = AppContext.TryGetSwitch("Npgsql.EnableLegacyTimestampBehavior", out var enabled) && enabled;
+        Diagnostics = AppContext.TryGetSwitch("Npgsql.EnableDiagnostics", out var enabled) && enabled;
```

Comment:
Just a suggestion.

Reply:
Yeah, not a fan of that specific naming - it's pretty ambiguous with diagnostics in the OpenTelemetry sense (e.g. [this](https://www.npgsql.org/doc/diagnostics/overview.html)).

EnableSafetyChecks (FWIW EF has EnableThreadSafetyChecks)? EnableExtraChecks? EnableExtraSafetyChecks? If we want to make it super narrow it can be EnableBoundsChecks, but it's nice to have a more general switch to cover possibly other stuff...

I'd personally go with EnableExtraChecks I think...

Reply:
What about EnableAssertions?

Reply:
Sure, sounds good.